lib/agent_runner.rb

```diff
@@ -58,9 +58,14 @@ def self.register(agent)
   end
 
   def self.with_connection
-    ActiveRecord::Base.connection_pool.with_connection do
-      yield
-    end
+    begin
+  	  ActiveRecord::Base.connection_pool.with_connection do
```

Comment:
Hey @jn11, the indentation seems off here.  It should be 2 spaces (not tabs) per block level.